framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/builder/ImmutableKieServerScenarioBuilderImpl.java

```diff
@@ -0,0 +1,206 @@
+
+        kieApp.getMetadata().setName(OpenShiftConstants.getKieApplicationName());
+        kieApp.getSpec().setEnvironment(OpenShiftOperatorEnvironments.PRODUCTION_IMMUTABLE);
+
+        OpenShiftOperatorConstants.getKieImageRegistryCustom().ifPresent(registry -> {
+            ImageRegistry imageRegistry = new ImageRegistry();
+            imageRegistry.setInsecure(true);
+            imageRegistry.setRegistry(registry);
+            kieApp.getSpec().setImageRegistry(imageRegistry);
+        });
+
+        CommonConfig commonConfig = new CommonConfig();
+        commonConfig.setAdminUser(DeploymentConstants.getWorkbenchUser());
+        commonConfig.setAdminPassword(DeploymentConstants.getWorkbenchPassword());
+        commonConfig.setServerPassword(DeploymentConstants.getKieServerPassword());
+        commonConfig.setControllerPassword(DeploymentConstants.getControllerPassword());
+        commonConfig.setMavenPassword(DeploymentConstants.getWorkbenchPassword());
+        kieApp.getSpec().setCommonConfig(commonConfig);
+
+        Server server = new Server();
+        server.addEnvs(authenticationEnvVars);
+        kieApp.getSpec().getObjects().addServer(server);
+    }
+
+    @Override
+    public ImmutableKieServerScenario build() {
+        return new ImmutableKieServerScenarioImpl(kieApp, deploySSO);
+    }
+
+    @Override
+    public ImmutableKieServerScenarioBuilder withExternalMavenRepo(String repoUrl, String repoUserName, String repoPassword) {
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            server.addEnv(new Env(ImageEnvVariables.EXTERNAL_MAVEN_REPO_URL, repoUrl));
+            server.addEnv(new Env(ImageEnvVariables.EXTERNAL_MAVEN_REPO_USERNAME, repoUserName));
+            server.addEnv(new Env(ImageEnvVariables.EXTERNAL_MAVEN_REPO_PASSWORD, repoPassword));
+        }
+        return this;
+    }
+
+    @Override
+    public ImmutableKieServerScenarioBuilder deploySso() {
+        deploySSO = true;
+        Server[] servers = kieApp.getSpec().getObjects().getServers();
+        for (int i=0; i<servers.length; i++) {
+            SsoClient ssoClient = new SsoClient();
+            ssoClient.setName("kie-server-" + i + "-client");
+            ssoClient.setSecret("kie-server-" + i + "-secret");
+            servers[i].setSsoClient(ssoClient);
+        }
+        return this;
+    }
+
+    @Override
+    public ImmutableKieServerScenarioBuilder withKieServerId(String kieServerId) {
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            server.addEnv(new Env(ImageEnvVariables.KIE_SERVER_ID, kieServerId));
+        }
+        return this;
+    }
+
+    @Override
+    public ImmutableKieServerScenarioBuilder withHttpKieServerHostname(String hostname) {
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            server.addEnv(new Env(ImageEnvVariables.HOSTNAME_HTTP, hostname));
```

Comment:
Shouldn't we add there a suffix to make a different hostnames for more servers? (In similar way how are resolved sso clients). Or is that add in operator, or it can be used only when is set only one Kie Server deployment?

Reply:
Well, in this scenario there is just one Kie server deployment available. I kept there for loop as kind of safety measure (or just a copy paste?).